cloudbuild_docker.yaml

```diff
@@ -42,6 +43,52 @@ steps:
     waitFor:
       - show-target-build-platforms
     id: 'build-witness-image'
+  # Write this build to a serverless log. Note that this step is experimental, and it is configured
+  # to always appear to succeed from the view of Cloud Build, although more details may be found in
+  # the logs.
+  - name: 'gcr.io/cloud-builders/docker'
+    entrypoint: bash
+    args:
+      # Set ERR trap to be inherited by shell functions. Allows our bash command logs to show up in
+      # Cloud Console.
+      - '-E'
+      - '-c'
+      - >
+        docker pull gcr.io/$PROJECT_ID/witness &&
+        echo $(docker images --format '{{.Digest}}' gcr.io/$PROJECT_ID/witness) > /workspace/image-digest.txt ||
+        # Tell Cloud Build that this step succeeded. See this workaround documented here:
+        # https://github.com/GoogleCloudPlatform/cloud-builders/issues/253#issuecomment-381600420
+        exit 0
+    id: 'store-witness-image-digest'
+    waitFor:
+      - build-witness-image
+  - name: 'gcr.io/cloud-builders/gcloud'
+    entrypoint: 'bash'
+    args:
+      # Set ERR trap to be inherited by shell functions. Allows our bash command logs to show up in
+      # Cloud Console.
+      - '-E'
+      - '-c'
+      - >
+        # Special /workspace path:
+        # https://cloud.google.com/build/docs/configuring-builds/pass-data-between-steps#passing_data_using_workspaces
+        gcloud functions call create_gcs_object --data="{
+          \"bucket\":\"${_BUCKET}\",
+          \"entryContent\":\"${REPO_NAME},${COMMIT_SHA},$(cat /workspace/image-digest.txt),linux/amd64\",
+          \"entryPath\":\"${REPO_NAME}-linux-amd64\"
+        }" &&
```

Comment:
FYI the quote escaping is a bit annoying here, but I couldn't find a way to avoid it. The outside double quotes allow the `$(cat ...)` command to be executed, and the inside/escaped double quotes are required by JSON.

Reply:
Yeah, it's ok I've been down this bumpy road too :)